packages/next/src/utils.ts

```diff
@@ -2429,7 +2429,8 @@ export const onPrerenderRoute =
                 initialHeaders: {
                   'content-type': rscContentTypeHeader,
                   vary: rscVaryHeader,
-                  ...(experimentalPPR && rscDidPostponeHeader
+                  // If it contains a pre-render, then it was postponed.
+                  ...(prerender && rscDidPostponeHeader
```

Comment:
This ensures that the prerender is only generated with the "didPostponeHeader" being set if indeed that specific prerender did actually postpone. This is indicated by the truthyness of the `prerender` variable here.